packages/axelar-soroban-std/src/interfaces/ownable.rs

```diff
@@ -0,0 +1,140 @@
+use crate::events::Event;
+#[cfg(any(test, feature = "testutils"))]
+use crate::impl_event_testutils;
+use core::fmt::Debug;
+use soroban_sdk::{contractclient, Address, Env, IntoVal, Symbol, Topics, Val, Vec};
+
+#[contractclient(name = "OwnableClient")]
+pub trait OwnableInterface {
+    /// Returns the address of the contract's owner.
+    fn owner(env: &Env) -> Address;
+
+    /// Transfers ownership of the contract to a new address.
+    fn transfer_ownership(env: &Env, new_owner: Address);
+}
+
+/// Default implementation of the [OwnableInterface] trait.
+pub fn owner(env: &Env) -> Address {
+    env.storage()
+        .instance()
+        .get(&storage::DataKey::Interfaces_Owner)
+        .expect("owner must be set during contract construction")
+}
+
+/// Default implementation of the [OwnableInterface] trait. Ensures the current owner is authorized and emits an event after the transfer.
+pub fn transfer_ownership<T: OwnableInterface>(env: &Env, new_owner: Address) {
+    let previous_owner = T::owner(env);
+    previous_owner.require_auth();
+
+    set_owner(env, &new_owner);
+
+    OwnerChangedEvent {
+        previous_owner,
+        new_owner,
+    }
+    .emit(env);
+}
+
+/// Default implementation accompanying the [OwnableInterface] trait. This should never be part of a contract interface,
+/// but allows contracts internally to set the owner.
+pub fn set_owner(env: &Env, owner: &Address) {
+    env.storage()
+        .instance()
+        .set(&storage::DataKey::Interfaces_Owner, owner);
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct OwnerChangedEvent {
+    pub previous_owner: Address,
+    pub new_owner: Address,
+}
+
+impl Event for OwnerChangedEvent {
+    fn topics(&self, env: &Env) -> impl Topics + Debug {
+        (
+            Symbol::new(env, "ownership_transferred"),
+            self.previous_owner.to_val(),
+            self.new_owner.to_val(),
+        )
+    }
+
+    fn data(&self, env: &Env) -> impl IntoVal<Env, Val> + Debug {
+        Vec::<Val>::new(env)
+    }
+}
+
+#[cfg(any(test, feature = "testutils"))]
+impl_event_testutils!(OwnerChangedEvent, (Symbol, Address, Address), ());
+
+// submodule to encapsulate the disabled linting
+mod storage {
+    // linting is disabled for the enum variant names on purpose, so we can define names that would otherwise be invalid.
```

Comment:
Maybe add an entry in the readme and link to it instead since we'll have this in a few places?